### PR TITLE
[Kotlin migration] MyAccount

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
Kotlin Migration

## Fixes
Fixes _Link to the issues._

## Approach
1. Kotlin migration standard procedure
2. Fix errors

## How Has This Been Tested?

Manually tried to make a login (Samsung Galaxy Note 10 Lite SM-N770F/DS, API 30)

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by BrayanDSO** [comment]
2 questions

1. [Code style](https://github.com/ankidroid/Anki-Android/wiki/Code-style) looks like is written only to Java. In kotlin do we still follow hungarian notation?
2. Lint thereatened me to remove unused variable `root` in order to compile. Should I have commented it or supressed?

https://github.com/ankidroid/Anki-Android/blob/81e770b843cf3e28ee447af619baf871c1a6172a/AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt#L233-L235

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 1]
```diff
@@ -11,373 +11,343 @@
...
-        String password = mPassword.getText().toString();
-
-        if (!"".equalsIgnoreCase(username) && !"".equalsIgnoreCase(password)) {
-            Timber.i("Attempting auto-login");
-            Connection.login(mLoginListener, new Connection.Payload(new Object[]{username, password,
-                    HostNumFactory.getInstance(this) }));
+    fun attemptLogin() {
+        val username = mUsername!!.text.toString().trim { it <= ' ' } // trim spaces, issue 1586
+        val password = mPassword.text.toString()
+        if (!"".equals(username, ignoreCase = true) && !"".equals(password, ignoreCase = true)) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:111]: Cleanup: Ignorecase seems useless here

I suspect we could get away with `username.isNotBlank() && password.isNotEmpty()`

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:111]: Cleanup: invert if & return early

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 2]
```diff
@@ -11,373 +11,343 @@
...
-
-    private void saveUserInformation(String username, String hkey) {
-        SharedPreferences preferences = AnkiDroidApp.getSharedPrefs(getBaseContext());
-        Editor editor = preferences.edit();
-        editor.putString("username", username);
-        editor.putString("hkey", hkey);
-        editor.apply();
+    private fun saveUserInformation(username: String, hkey: String) {
+        val preferences = AnkiDroidApp.getSharedPrefs(baseContext)
+        val editor = preferences.edit()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:129]: cleanup: `preferences.edit { }`

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 3]
```diff
@@ -11,373 +11,343 @@
...
-
+            mPassword.error = getString(R.string.password_empty)
+            mPassword.requestFocus()
+            return
         }
-
-        if (!"".equalsIgnoreCase(username) && !"".equalsIgnoreCase(password)) {
-            Connection.login(mLoginListener, new Connection.Payload(new Object[]{username, password,
-                    HostNumFactory.getInstance(this) }));
+        if (!"".equals(username, ignoreCase = true) && !"".equals(password, ignoreCase = true)) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:151]: ditto - we should also extract the duplication into a method

Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:151]: I also think this is logically impossible given the above code

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 4]
```diff
@@ -11,373 +11,343 @@
...
-    private void logout() {
-        SharedPreferences preferences = AnkiDroidApp.getSharedPrefs(getBaseContext());
-        Editor editor = preferences.edit();
-        editor.putString("username", "");
-        editor.putString("hkey", "");
-        editor.apply();
-        HostNumFactory.getInstance(this).reset();
+    private fun logout() {
+        val preferences = AnkiDroidApp.getSharedPrefs(baseContext)
+        val editor = preferences.edit()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:168]: `.edit {}`

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 5]
```diff
@@ -11,373 +11,343 @@
...
-        lostEmail.setOnClickListener(v -> openUrl(lostMailUrl));
-
-
-        mLoggedIntoMyAccountView = getLayoutInflater().inflate(R.layout.my_account_logged_in, null);
-        mUsernameLoggedIn = mLoggedIntoMyAccountView.findViewById(R.id.username_logged_in);
-        Button logoutButton = mLoggedIntoMyAccountView.findViewById(R.id.logout_button);
-        logoutButton.setOnClickListener(v -> logout());
-
+        )
+        val loginButton = mLoginToMyAccountView.findViewById<Button>(R.id.login_button)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:202]: ⚠️ I'd prefer if the spacing here was reverted to how it was in the Java

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:202]: cleanup: we can then use scope functions

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 6]
```diff
@@ -11,373 +11,343 @@
...
+            if (loginMessage == null || loginMessage.length == 0) {
                 if (messageResource == R.string.youre_offline && !Connection.getAllowLoginSyncOnNoConnection()) {
-                    //#6396 - Add a temporary "Try Anyway" button until we sort out `isOnline`
-                    View root = this.findViewById(R.id.root_layout);
-                    UIUtils.showSnackbar(this, messageResource, false, R.string.sync_even_if_offline, (v) -> {
-                        Connection.setAllowLoginSyncOnNoConnection(true);
-                        login();
-                    }, null);
+                    // #6396 - Add a temporary "Try Anyway" button until we sort out `isOnline`
+//                    val root = this.findViewById<View>(R.id.root_layout)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:234]: ⚠️ move the comment start to be inline with the code

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 7]
```diff
@@ -11,373 +11,343 @@
...
-        @Override
-        public void onPreExecute() {
-            Timber.d("loginListener.onPreExecute()");
-            if (mProgressDialog == null || !mProgressDialog.isShowing()) {
-                mProgressDialog = StyledProgressDialog.show(MyAccount.this, null,
-                        getResources().getString(R.string.alert_logging_message), false);
+        override fun onPreExecute() {
+            Timber.d("loginListener.onPreExecute()")
+            if (mProgressDialog == null || !mProgressDialog!!.isShowing) {
+                mProgressDialog = show(
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:260]: ⚠️ remove the import to `StyledProgressDialog` and use `StyledProgressDialog.show()`

**Review by david-allison — CHANGES_REQUESTED** [review]
Looks great!

* one spacing fix
* One import was removed 
* One comment tabbing issue

**Comment by david-allison** [comment]
1. Comment the line for now, to ensure a nicer diff between Java and Kotlin

For now, keep the variable style, UNLESS `VariableNamingDetector` flags it up.

After the Kotlin Migration is completed, I hope to remove all Hungarian notation from the code, we should be able to flag this via lint once everything is in Kotlin.

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt (hunk 3 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:154]: True. aec219d7934473553a7e8a333f3a0716c8e5d0c7 superseded the check, but didn't remove the old one

**Review by david-allison — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]
A lot of cleanups :sweat_smile: - but nothing I can pick on at this point on the straight migration

**Pull request merged**